tools/nnyeah/nnyeah/AssemblyComparator/ComparingVisitor.cs

```diff
@@ -0,0 +1,144 @@
+			PublicOnly = publicOnly;
+		}
+
+		public void Visit ()
+		{
+			var earlierElements = ModuleElements.Import (EarlierModule, PublicOnly);
+			if (earlierElements is null)
+				throw new Exception (Errors.E0007);
+			var laterElements = ModuleElements.Import (LaterModule, PublicOnly);
+			if (laterElements is null)
+				throw new Exception (Errors.E0007);
+			var reworker = new TypeReworker (EarlierModule);
+			VisitTypes (reworker, earlierElements, laterElements);
+		}
+
+		void VisitTypes (TypeReworker reworker, ModuleElements earlier, ModuleElements later)
+		{
+			foreach (var typeName in earlier.Types.Keys) {
+				if (!later.Types.TryGetValue (typeName, out var laterElems)) {
+					TypeNotFound.Invoke (this, new TypeNotFoundEventArgs (typeName));
+					continue;
+				} else {
+					TypeFound.Invoke (this, new TypeFoundEventArgs (typeName, laterElems.DeclaringType.ToString ()));
+				}
+				if (!earlier.Types.TryGetValue (typeName, out var earlierElems)) {
+					throw new Exception (Errors.E0007);
+				}
+				VisitAllMembers (reworker, earlierElems, laterElems);
+			}
+		}
+
+		void VisitAllMembers (TypeReworker reworker, TypeElements earlier, TypeElements later)
+		{
+			VisitMembers (reworker, earlier.Methods, later.Methods);
+			VisitMembers (reworker, earlier.Fields, later.Fields);
+			VisitMembers (reworker, earlier.Events, later.Events);
+			VisitMembers (reworker, earlier.Properties, later.Properties);
+		}
+
+		void VisitMembers<T> (TypeReworker reworker,
+			List<TypeElement<T>> earlier, List<TypeElement<T>> later) where T : IMemberDefinition
+		{
+			foreach (var earlierElem in earlier) {
+				VisitMember (reworker, earlierElem, later);
+			}
+		}
+
+		void VisitMember<T> (TypeReworker reworker, TypeElement<T> elem,
+			List<TypeElement<T>> later) where T : IMemberDefinition
+		{
+			foreach (var late in later) {
+				if (elem.Signature == late.Signature) {
+					FireElementFound (elem, late);
+					return;
+				}
+			}
+			var remappedSig = RemappedSignature (reworker, elem.Element);
+			if (remappedSig == elem.Signature) {
+				FireElementNotFound (elem);
+				return;
+			}
+			foreach (var late in later) {
+				if (remappedSig == late.Signature) {
+					FireElementFound (elem, late);
+					return;
+				}
+			}
+			FireElementNotFound (elem);
+		}
+
+		void FireElementNotFound<T> (TypeElement<T> earlier) where T : IMemberDefinition
+		{
+			if (earlier is TypeElement<FieldDefinition> field) {
+				FieldNotFound.Invoke (this, new FieldNotFoundEventArgs (field.Signature));
+			} else if (earlier is TypeElement<MethodDefinition> method) {
+				MethodNotFound.Invoke (this, new MethodNotFoundEventArgs (method.Signature));
+			} else if (earlier is TypeElement<EventDefinition> @event) {
+				EventNotFound.Invoke (this, new EventNotFoundEventArgs (@event.Signature));
+			} else if (earlier is TypeElement<PropertyDefinition> property) {
+				PropertyNotFound.Invoke (this, new PropertyNotFoundEventArgs (property.Signature));
```

Comment:
In RemappedSignature you switch on types, and throw if it is unexpected, which is superior to this form.